examples/inspection/plot_linear_model_coefficient_interpretation.py

```diff
@@ -0,0 +1,602 @@
+    columns=feature_names
+)
+plt.figure(figsize=(9, 7))
+sns.swarmplot(data=coefs, orient='h', color='k', alpha=0.5)
+sns.boxplot(data=coefs, orient='h', color='cyan', saturation=0.5)
+plt.axvline(x=0, color='.5')
+plt.title('Coefficient variability')
+plt.subplots_adjust(left=.3)
+
+##############################################################################
+# The result is quite similar to the non-normalized case.
+#
+# Linear models with regularization
+# ---------------------------------
+#
+# In machine-learning practice, Ridge Regression is more often used with
+# non-negligible regularization.
+# Above, we limited this regularization to a very little amount.
+# Regularization improves the conditioning of the problem and reduces the
+# variance of the estimates. RidgeCV applies cross validation in order to
+# determine which value of the regularization parameter (`alpha`) is best
+# suited for the model estimation.
+
+from sklearn.linear_model import RidgeCV
+
+model = make_pipeline(
+    preprocessor,
+    TransformedTargetRegressor(
+        regressor=RidgeCV(alphas=np.logspace(-10, 10, 21)),
+        func=np.log10,
+        inverse_func=sp.special.exp10
+    )
+)
+
+_ = model.fit(X_train, y_train)
+
+##############################################################################
+# First we verify which value of :math:`\alpha` has been selected.
+
+model[-1].regressor_.alpha_
+
+##############################################################################
+# Then we check the quality of the predictions.
+
+y_pred = model.predict(X_train)
+mae = median_absolute_error(y_train, y_pred)
+string_score = 'MAE on training set: {0:.2f} $/hour'.format(mae)
+y_pred = model.predict(X_test)
+mae = median_absolute_error(y_test, y_pred)
+string_score += '\nMAE on testing set: {0:.2f} $/hour'.format(mae)
+
+fig, ax = plt.subplots(figsize=(6, 6))
+plt.scatter(y_test, y_pred)
+ax.plot([0, 1], [0, 1], transform=ax.transAxes, ls="--", c="red")
+
+plt.text(3, 20, string_score)
+
+plt.title('Ridge model, regularization, normalized variables')
+plt.ylabel('Model predictions')
+plt.xlabel('Truths')
+plt.xlim([0, 27])
+plt.ylim([0, 27])
+
+##############################################################################
+# The R squared coefficient is similar to the non-regularized case.
+
+coefs = pd.DataFrame(
+    model.named_steps['transformedtargetregressor'].regressor_.coef_,
+    columns=['Coefficients'], index=feature_names
+)
+coefs.plot(kind='barh', figsize=(9, 7))
+plt.title('Ridge model, regularization, normalized variables')
+plt.axvline(x=0, color='.5')
+plt.subplots_adjust(left=.3)
+
+##############################################################################
+# Coefficients are significantly different.
+# AGE and EXPERIENCE coefficients are both positive.
+# Even if the model is still not able to provide a good description of the
+# dataset, the regularization manages to lower the influence of correlated
+# variables on the model.
```

Comment:
I'm not sure if you want to say that even if their sign is the same, you still can't interpret the coefficient of AGE without appreciating its correlation with EXPERIENCE!

Reply:
I would like to point out the following.
Knowing that EXPERIENCE and AGE are correlated (see the pairwise plot at the beginning of the example), the way RidgeCV models with bigger regularization take account of that is minimizing the correspondent coefficients, while making them more robust with respect to cross validation. This is different from lasso models (for example) which drop one of the correlated variables.
Is that correct? I understand I should reformulate...  :)

Reply:
Lasso models also do not necessarily drop one of the correlated variables (it is not minimising the number of variables in the model, only the sum of their magnitudes), but ridge tends to put nonzero weight on all variables that may help to predict the target.

I think my point here is that if I took a list of the most predictive variables, AGE and EXPERIENCE might not be among them, because Ridge would share the weight between the two predictive variables, so neither alone would be very strongly weighted.

Reply:
@jnothman , thanks for your comments, they help me to understand what I'm doing here! :)
I have added more details in this section. If you don't mind, I'm going to resolve this conversation just for a readability concern. Feel free to comment in the new version. Thanks!